packages/amazon-bedrock/src/bedrock-chat-language-model.ts

```diff
@@ -377,19 +387,111 @@ export class BedrockChatLanguageModel implements LanguageModelV1 {
               }
             }
           },
-
           flush(controller) {
             controller.enqueue({
               type: 'finish',
               finishReason,
               usage,
-              providerMetadata,
+              ...(providerMetadata && { providerMetadata }),
             });
           },
         }),
       ),
       rawCall: { rawPrompt, rawSettings },
+      rawResponse: { headers: responseHeaders },
       warnings,
     };
   }
+
+  private getUrl(modelId: string) {
+    const encodedModelId = encodeURIComponent(modelId);
+    return `${this.config.baseUrl()}/model/${encodedModelId}/converse`;
+  }
+
+  private getStreamUrl(modelId: string): string {
+    const encodedModelId = encodeURIComponent(modelId);
+    return `${this.config.baseUrl()}/model/${encodedModelId}/converse-stream`;
+  }
 }
+
+const BedrockStopReasonSchema = z.union([
+  z.enum(BEDROCK_STOP_REASONS),
+  z.string(),
+]);
+
+// limited version of the schema, focussed on what is needed for the implementation
+// this approach limits breakages when the API changes and increases efficiency
+const BedrockResponseSchema = z.object({
+  metrics: z
+    .object({
+      latencyMs: z.number(),
+    })
+    .nullish(),
+  output: z.object({
+    message: z.object({
+      content: z.array(
+        z.object({
+          text: z.string().optional(),
+          toolUse: z
+            .object({
+              toolUseId: z.string(),
+              name: z.string(),
+              input: z.any(),
+            })
+            .optional(),
+        }),
+      ),
+      role: z.string(),
+    }),
+  }),
+  stopReason: BedrockStopReasonSchema,
+  trace: z.any().nullish(),
+  usage: z.object({
+    inputTokens: z.number(),
+    outputTokens: z.number(),
+    totalTokens: z.number(),
+  }),
+});
+
+// limited version of the schema, focussed on what is needed for the implementation
+// this approach limits breakages when the API changes and increases efficiency
+const BedrockStreamSchema = z.object({
+  contentBlockDelta: z
+    .object({
+      contentBlockIndex: z.number(),
+      delta: z.record(z.any()).nullish(),
```

Comment:
We should avoid `any` in the response parsing. If we don't want to go into details, we can use `z.unknown` afaik